packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker.tsx

```diff
@@ -76,6 +77,9 @@ export function TimeRangePicker(props: TimeRangePickerProps) {
   };
 
   const onToolbarButtonSwitch = () => {
+    if (!isOpen && props.onToolbarTimePickerClick) {
+      props.onToolbarTimePickerClick();
+    }
     setOpen((prevState) => !prevState);
```

Comment:
Since state updates are async and could lead to unexpected behaviors, I would put that if statement inside `setOpen` function in order to always use `prevState` instead of `isOpen` value.

```suggestion
  setOpen((prevState) => {
    if (prevState && props.onToolbarTimePickerClick) {
      props.onToolbarTimePickerClick();
    }
    return !prevState;
  });
```

Another way, I think cleaner than this one, is to use a useEffect hook with isOpen dependency.